pkg/controller/provider/container/ovirt/client.go

```diff
@@ -219,3 +219,156 @@ func GetInsecureSkipVerifyFlag(secret *core.Secret) bool {
 
 	return insecureSkipVerify
 }
+
+func TestDownloadOvfStore(secret *core.Secret, log logr.Logger) (ok bool, err error) {
+
+	// Create the connection to the server:
+	conn, err := ovirtsdk.NewConnectionBuilder().
+		URL(string(secret.Data["url"])).
+		Username(string(secret.Data["user"])).
+		Password(string(secret.Data["password"])).
+		Insecure(true).
+		Compress(true).
+		Timeout(time.Second * 30).
+		Build()
+	if err != nil {
+		log.Error(err, "Failed establishing connection to oVirt")
+		err = liberr.Wrap(err)
+		return
+	}
+	defer conn.Close()
+
+	//Find OVF STORE ID
+	disksService := conn.SystemService().DisksService()
+	diskResponse, err := disksService.List().Send()
+	if err != nil {
+		log.Error(err, "Failed to get disks list from oVirt")
+		err = liberr.Wrap(err)
+		return
+	}
+
+	var ovfStoreId string
+	if disks, ok := diskResponse.Disks(); ok {
+		//get OVF STORE ID
+		for _, disk := range disks.Slice() {
+			if diskName, ok := disk.Name(); ok {
+				if diskName == "OVF_STORE" {
+					if diskId, ok := disk.Id(); ok {
+						ovfStoreId = diskId
+					}
+					break
+				}
+			}
+		}
+	}
+	// In case no OVF store was found in the cluster
+	if ovfStoreId == "" {
+		log.Error(err, "OVF Store does not exist on cluster, CA certificate test can't performed")
+		ok = false
+		return
+	}
+
+	//in case the OVF STORE is locked wait fo OK status before stating the image transfer
+	conn.WaitForDisk(ovfStoreId, ovirtsdk.DISKSTATUS_OK, 20*time.Second)
+
+	// Prepare image transfer request
+	transfersService := conn.SystemService().ImageTransfersService()
+	transfer := transfersService.Add()
+	image, err := ovirtsdk.NewImageBuilder().Id(ovfStoreId).Build()
+	if err != nil {
+		log.Error(err, "Failed to build image for image transfer")
+		err = liberr.Wrap(err)
+		return
+	}
+	imageTransfer, err := ovirtsdk.NewImageTransferBuilder().Image(
+		image,
+	).Direction(
+		ovirtsdk.IMAGETRANSFERDIRECTION_DOWNLOAD,
+	).Build()
+	if err != nil {
+		log.Error(err, "Failed to build image transfer")
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Initialize image transfer and lock the disk
+	transfer.ImageTransfer(imageTransfer)
+	transferReq, err := transfer.Send()
```

Comment:
no need, changing into one.
regarding the name I've used req because it's the request to initiate image transfer, I saw they were using similar names in the ovirt SDK flow, but I can change it to `response` if you think it's more suitable.

Reply:
I see, but here it seems to be a response of the request to start a transfer, no? if so then yeah, let's change it so it will better reflect what it represents